log/logger.go

```diff
@@ -31,21 +31,23 @@ func NewNullLogger() *Logger {
 }
 
 // New creates a new logger.
-func New(logger *logrus.Logger, iterID string) *Logger {
-	var defLogger bool
+func New(logger logrus.FieldLogger, iterID string) *Logger {
+	var ll *logrus.Logger
+
 	if logger == nil {
-		defLogger = true
-		logger = logrus.New()
+		ll = logrus.New()
+		ll.Warn("Logger", "no logger supplied, using default")
+	} else if l, ok := logger.(*logrus.Logger); ok {
+		ll = l
+	} else {
+		// panic so that we know when to upgrade. see issue #818.
+		panic("logger is not a logrus.Logger")
```

Comment:
Could we work with the default (create a new logger like we do in the `nil` case) logger instead of panicking?

Reply:
Sure, but does it provide the same kind of security? I fear we might easily miss it.

Reply:
Ah, I see what you mean. You want to keep working with the `*logrus.Logger` no matter what. That's also an option, yes. WDYT? @na--?

Reply:
Yeah, that makes sense. Could you try working with `ErrFatal` instead of `panic`?

Reply:
@ankur22 Please take a look: https://github.com/grafana/xk6-browser/pull/824/commits/3b2de1b241017bad1707289ca974a7264be3463b.